opencti-platform/opencti-graphql/src/modules/auth/auth-domain.ts

```
@@ -0,0 +1,139 @@
+  if (user.external) throw UnsupportedError('External user');
+  return user;
+};
+
+export const generateOtp = () => {
+  let otp = '';
+  for (let i = 0; i < 8; i += 1) {
+    const random = Math.floor(Math.random() * 10);
+    otp += random;
+  }
+  return otp;
+};
+
+interface SendMailArgs {
+  from: string;
+  to: string;
+  subject: string;
+  html: string;
+}
+
+export const askSendOtp = async (context: AuthContext, input: AskSendOtpInput) => {
+  const settings = await getEntityFromCache<BasicStoreSettings>(context, ADMIN_USER, ENTITY_TYPE_SETTINGS);
+  const resetOtp = generateOtp();
+  try {
+    const user = await getUser(input.email);
+    const { user_email, name } = user;
+    const email = user_email.toLowerCase();
+    const storedOtp = await redisGetForgotPasswordOtp(input.email);
+    // Prevent code generation if generated less than 30 seconds ago
+    const isTooRecentStoredOtp = storedOtp.ttl > (OTP_TTL - 30);
+    if (isTooRecentStoredOtp) return true;
+    await redisSetForgotPasswordOtp(email, resetOtp);
+    const body = `Hi ${name},</br>`
+        + 'A request has been made to reset your OpenCTI password.</br></br>'
+        + 'Enter the following password recovery code:</br></br>'
+        + `<b>${resetOtp}</b>`;
+
+    const sendMailArgs: SendMailArgs = {
+      from: settings.platform_email,
+      to: user_email,
+      subject: `${resetOtp} is your recovery code of your OpenCTI account`,
+      html: ejs.render(OCTI_EMAIL_TEMPLATE, { settings, body }),
+    };
+    await sendMail(sendMailArgs);
+    await publishUserAction({
+      user: SYSTEM_USER,
+      event_type: 'authentication',
+      event_scope: 'forgot',
+      event_access: 'administration',
+      context_data: undefined,
+      message: `send an OTP to ${user_email}`,
+    });
+  } catch (e) {
+    // Prevent wrong email address, but return true too if it fails
+    // logApp.error('Error occurred while sending password reset email:', { cause: e });
```

Why is this error commented ? It's better to not have silent errors.

---

It's an oversight